### PR TITLE
Add snapshot test to verify all commands load properly

### PR DESCRIPTION
**Pull request by Arkham**
### WHY are these changes introduced?

We recently broke cli ([here](https://github.com/Shopify/cli/pull/494) and [here](https://github.com/Shopify/cli/pull/528)) by messing up the imports.

This PR introduces a simple [Golden test](https://ro-che.info/articles/2017-12-04-golden-tests) to ensure that all commands that we expect to be present DO load correctly.

### WHAT is this pull request doing?

It creates a snapshot of loaded CLI commands inside the `packages/features/snapshots/commands.txt` file.

### How to test your changes?

- Cd into `packages/features`
- Run `FEATURE=features/commands.feature yarn test`, it should say all good
- Change `./snapshots/commands.txt` and add a new line, i.e. `app fun                @shopify/app`
- Run `FEATURE=features/commands.feature yarn test` again and it should complain in this fashion:
```
1) Scenario: No runtime errors happen when loading the CLI modules graph # features/commands.feature:3
   ✖ Then I see all available commands and they match the snapshot # steps/commands.steps.ts:24
       AssertionError [ERR_ASSERTION]: 
       SNAPSHOT TEST FAILED!
       
       The result of 'yarn shopify commands' has changed! We run this to check that
       all commands can load successfully.
       
       It's normal to see this test fail when you add or remove a command in the CLI.
       In this case you can run this command to regenerate the snapshot file:
       
       $ yarn test:regenerate-snapshots
       
       Then you can commit this change and this test will pass.
       
       If instead you didn't mean to change a command, UH OH. Check the commands in
       the diff below and figure out what is broken.
       
           + expected - actual
       
             app build              @shopify/app           
             app deploy             @shopify/app           
           - app fun                @shopify/app           
             app dev                @shopify/app           
             app env pull           @shopify/app           
             app env show           @shopify/app           
             app generate extension @shopify/app
```
- also test that running `yarn test:regenerate-snapshots` does not create a diff 🙏 

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by Arkham** [comment]
You can also try by reverting one of the fixes to the issues posted above and verify that it would have picked up the breakage:

- `git revert 3c59632acd5208068e836c383bbc3ae8c7c364d3`
- `./golden/run-tests.sh`

File: .github/workflows/shopify-cli.yml [hunk 1]
```diff
@@ -220,3 +222,5 @@ jobs:
       - name: Acceptance tests
         if: ${{ matrix.node == '18.7.0' }}
         run: yarn nx run features:test
+      - name: Golden tests
```
Comment (pepicrft) [inline .github/workflows/shopify-cli.yml:225]: What about building it upon the existing Cucumber setup?

```ghekin
Scenario: No runtime errors happen when loading the modules graph
  Then: I run help for every command available
```
In my experience working with bash scripts is usually friction when contributing.

File: packages/features/steps/commands.steps.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Then you can commit this change and this test will pass.
+
+If instead you didn't mean to change a command, UH OH. Check the commands in
+the diff below and figure out what is broken.
+`
+
+Then(/I see all available commands and they match the snapshot/, async function () {
+  const commandFlags = ['commands', '--no-header', '--columns=Command,Plugin']
+  const result: string = (
+    await exec('node', [executables.cli, ...commandFlags], {env: {...process.env, ...this.temporaryEnv}})
```
Comment (matteodepalo) [inline packages/features/steps/commands.steps.ts:27]: 👍

**Review by matteodepalo — APPROVED** [review]

File: packages/features/features/commands.feature [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Feature: Loading all commands
+
+Scenario: No runtime errors happen when loading the CLI modules graph
+   Then I see all available commands and they match the snapshot
```
[the remark this answers is not in the document]
Comment (Arkham) [reply packages/features/features/commands.feature:4]: yup, good point, I'll make the change!

**Pull request merged**